ptbcontrib/username_to_chat_api/usernametochatapi.py

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env python
+#
+# A library containing community-based extension for the python-telegram-bot library
+# Copyright (C) 2020-2021
+# The ptbcontrib developers
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser Public License for more details.
+#
+# You should have received a copy of the GNU Lesser Public License
+# along with this program.  If not, see [http://www.gnu.org/licenses/].
+"""
+This module contains a class that, once initiated, works as a shortcut to the UsernameToChatAPI
+and puts the response in a Chat object, as well as puts the error to the fitting TelegramErrors.
+"""
+
+try:
+    import telegram.vendor.ptb_urllib3.urllib3 as urllib3
+except ImportError:  # pragma: no cover
+    import urllib3
+
+try:
+    import ujson as json
+except ImportError:  # pragma: no cover
+    import json  # type: ignore[no-redef]
+
+from telegram import error, Chat, Bot
+
+
+class UsernameToChatAPI:  # pylint: disable=too-few-public-methods
+    """
+    This class stores the URL and api key for a UsernameToChatAPI instance. Then, the correct
+    request is being made and the Chat returned.
+
+     Args:
+         api_url (:obj:`str`): The base URL (speak: domain) to the API.
+         api_key (:obj:`str`): The key, necessary to access the API.
+         bot (:obj:`telegram.Bot`): An initiated bot object, to create the Chat object.
+         pool_manager (:obj:`urllib3.PoolManager`, optional): Pre initialized
+             :obj:`urllib3.PoolManager`.
+    """
+
+    def __init__(
+        self, api_url: str, api_key: str, bot: Bot, pool_manager: urllib3.PoolManager = None
+    ) -> None:
+        if api_url.endswith("/"):
+            api_url = api_url[:-1]
+        self._url = api_url + "/resolveUsername"
+        self._api_key = api_key
+        self._bot = bot
+        if pool_manager:
+            self._http = pool_manager
+        else:
+            self._http = urllib3.PoolManager()
+
+    def resolve(self, username: str) -> Chat:
+        """
+        Returns the Chat object for an username.
+        """
+        response = self._http.request(
+            'GET', self._url, fields={"api_key": self._api_key, "username": username}
+        )
+        result = json.loads(response.data.decode('utf-8'))
+        if result["ok"]:
```

Comment:
isn't there `response.{ok, status}`? I'd vote to use the dot notations everywhere …

Reply:
no, its a dict. json yields a dict, not an class

Reply:
oh but ok is the thing in the json response. like telegram does it.

Reply:
mh, fair point. I just know that we use `resp.status` in ptb-request, which is cleaner imo, but then it should be checked before checking `reult['ok']`.

Reply:
`AttributeError: 'HTTPResponse' object has no attribute 'ok'`

I tried it, doesn't work; and I think it will also be okay if the json "ok" is False, because the request itself is successful.